R/content.R

```diff
@@ -925,3 +975,57 @@ get_content_permissions <- function(content, add_owner = TRUE) {
   res <- content$permissions(add_owner = add_owner)
   parse_connectapi_typed(res, connectapi_ptypes$permissions)
 }
+
+#' Render a content item.
+#' 
+#' @description Submit a request to render a content item. Once submitted, the
+#' server runs an asynchronous process to render the content. This might be
+#' useful if content needs to be updated after its source data has changed,
+#' especially if this doesn't happen on a regular schedule.
+#' 
+#' Only valid for rendered content (e.g., most Quarto documents, Jupyter
+#' notebooks, R Markdown reports).
+#' 
+#' @param content The content item you wish to render.
+#' @return A [VariantTask] object that can be used to track completion of the render.
+#' @export
+content_render <- function(content) {
```

Comment:
Would it be useful to add `variant` to the signature, like this, and then `get_variant()` in the function? 

```suggestion
content_render <- function(content, variant = "default") {
```

Reply:
I… don't have any objection to this! @tdstein Does Python's `render` function also let you target variants?

Reply:
Right now this function returns a `ContentTask` type. If this allowed arbitrary variants, it'd have to return a `VariantTask` type till I fix the weird typing around variants. 🤔

Reply:
No, the Variants API bindings are not yet public in the Python implementation. But, the eventual implementation will look like:

*via the content resource*

```py
from posit import connect

client = connect.Client()
content = client.content.get(...)
task = content.render(variant_id=...)
```

and 

*via the variant resource*

```py
from posit import connect

client = connect.Client()
content = client.content.get(...)
variant = content.variants.get(...)
task = variant.render()
```